### PR TITLE
feat(pie-textarea): DSW-2136 add disabled prop

### PR DESCRIPTION
**Pull request by leksaBoiko**
## Describe your changes (can list changeset entries if preferable)

**[Added]** - Render a textarea inside the pie-textarea with a `disabled` property that defaults to `false`

## Author Checklist (complete before requesting a review)
- [x] I have performed a self-review of my code
- [x] I have added thorough tests where applicable (component / visual)
- [x] I have reviewed the `PIE Storybook` PR preview
- [x] I have reviewed visual test updates properly before approving
- [x] If changes will affect consumers of the package, I have created a changeset entry.

### Reviewer 1 - @xander-marjoram 
- [x] I have reviewed the `PIE Storybook` PR preview
- [x] If there are visual test updates, I have reviewed them

### Reviewer 2
- [x] I have reviewed the `PIE Storybook` PR preview
- [x] If there are visual test updates, I have reviewed them

### CODE REVIEW

File: packages/components/pie-textarea/src/index.ts [hunk 1]
```diff
@@ -13,8 +16,21 @@ const componentSelector = 'pie-textarea';
...
     render () {
-        return html`<h1 data-test-id="pie-textarea">Hello world!</h1>`;
+        const {
+            disabled,
+        } = this;
+
+        return html`
+            <textarea
+                data-test-id="pie-textarea"
+                ?disabled=${live(disabled)}
```
Comment (kevinrodrigues) [inline packages/components/pie-textarea/src/index.ts:32]: Is `live` intended here are there updates outside of lit that can't pick this change up out of interest?

File: packages/components/pie-textarea/README.md [hunk 2]
```diff
@@ -73,8 +73,8 @@ import { PieTextarea } from '@justeattakeaway/pie-textarea/dist/react';
 ## Props
 
 | Property | Type | Default | Description |
-|---|---|---|---|
-| - | - | - | - |
+|--|--|--|--|
+| `disabled` | `boolean` | `false` | Indicates whether or not the checkbox is disabled. |
```
Comment (xander-marjoram) [inline packages/components/pie-textarea/README.md:77]:
```suggestion
| `disabled` | `boolean` | `false` | Indicates whether or not the textarea is disabled. |
```

File: packages/components/pie-textarea/test/visual/pie-textarea.spec.ts [hunk 3]
```diff
@@ -1,14 +1,61 @@
...
+        const propKeyValues = `disabled: ${testComponent.propValues.disabled}`;
+
+        await mount(
+            WebComponentTestWrapper,
+            {
+                props: { propKeyValues },
+            },
+        );
+
+        await percySnapshot(page, `PIE Textarea - Content and props - ${dir}`, percyWidths);
```
Comment (xander-marjoram) [inline packages/components/pie-textarea/test/visual/pie-textarea.spec.ts:58]: Looking at Percy, it looks like we only have screenshots for `disabled: true`, please can you add some for the `false` condition too?

File: packages/components/pie-textarea/test/component/pie-textarea.spec.ts [hunk 4]
```diff
@@ -17,4 +24,62 @@ test.describe('PieTextarea - Component tests', () => {
...
+                    // Act
+                    const textarea = page.locator('pie-textarea');
+                    await textarea.focus();
+
+                    // Assert
+                    expect(textarea).not.toBeFocused();
+                });
+            });
+
+            test.describe('when not provided', () => {
```
Comment (kevinrodrigues) [inline packages/components/pie-textarea/test/component/pie-textarea.spec.ts:59]: Might be worth updating this to match the one above:

`when false`

**Review by kevinrodrigues — APPROVED** [review]

**Review by xander-marjoram — APPROVED** [review]

**Pull request merged**